src/executables/loadjava.cpp

```diff
@@ -0,0 +1,239 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+/*
+ * loadjava.cpp - loadjava utility
+ */
+
+#ident "$Id$"
+
+#include "config.h"
+
+#include <cassert>
+#include <string>
+#include <regex>
+#include <filesystem>
+
+#include "cubrid_getopt.h"
+#include "error_code.h"
+#include "message_catalog.h"
+#include "utility.h"
+#include "databases_file.h"
+#if defined(WINDOWS)
+#include "porting.h"
+#endif /* WINDOWS */
+
+namespace fs = std::filesystem;
+using namespace std::regex_constants;
+
+#define JAVA_DIR                "java"
+#if defined(WINDOWS)
+#define SEPERATOR               "\\"
+#else /* ! WINDOWS */
+#define SEPERATOR               "/"
+#endif /* !WINDOWS */
+
+static const std::string JAVA_PACKAGE_PATTERN = "^([a-z_]{1}[a-z0-9_]*(\\.[a-z_]{1}[a-z0-9_]*)*)$";
+static const std::string SEPARATOR_STRING (SEPERATOR);
+
+static char *Program_name = NULL;
+static char *Dbname = NULL;
+std::string Src_class;
+std::regex Java_package_reg (JAVA_PACKAGE_PATTERN, ECMAScript | icase | optimize);
+
+static int Force_overwrite = false;
+static std::string package_path;
+
+static void
+usage (void)
+{
+  fprintf (stderr, "Usage: loadjava [OPTION] database-name java-class-file\n");
+  fprintf (stderr, "Options:\n-y\t%s\n-p\t%s\n",
+	   msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_LOADJAVA, LOADJAVA_ARG_FORCE_OVERWRITE_HELP),
+	   msgcat_message (MSGCAT_CATALOG_UTILS, MSGCAT_UTIL_SET_LOADJAVA, LOADJAVA_ARG_FORCE_PACKAGE_HELP));
+}
+
+static int
+parse_argument (int argc, char *argv[])
+{
+  struct option loadjava_option[] =
+  {
+    {"overwrite", 0, 0, 'y'},
+    {"package", 0, 0, 'p'},
+    {0, 0, 0, 0}
+  };
+
+  while (1)
+    {
+      int option_index = 0;
+      int option_key = getopt_long (argc, argv, "y:p:h", loadjava_option, &option_index);
+      if (option_key == -1)
+	{
+	  break;
+	}
+
+      switch (option_key)
+	{
+	case 'y':
+	  Force_overwrite = true;
+	  break;
+	case 'p':
+	{
+	  // check valid package name
+	  std::string package_name (optarg);
+	  if (!package_name.empty())
+	    {
+	      bool is_matched = std::regex_search (package_name, Java_package_reg);
+	      if (!is_matched)
+		{
+		  fprintf (stderr, "invalid java package name\n");
+		  return ER_FAILED;
+		}
+	      // replace all for package name's dot to SEPARATER
+	      // e.g. org.cubrid.abc => /org/cubrid/abc/
+	      package_path = SEPERATOR;
+	      package_path += std::regex_replace (package_name, std::regex ("\\."), SEPARATOR_STRING);
+	      package_path += SEPERATOR;
+	    }
+	}
+	break;
+	case 'h':
+	/* fall through */
+	default:
+	  usage ();
+	  return ER_FAILED;
+	}
+    }
+
+  if (optind + 1 < argc)
+    {
+      Dbname = argv[optind];
+      Src_class = argv[optind + 1];
```

Comment:
It will be helpful if we can handle many src classes at once, because compiling a java file of a PL/CSQL program usually generates many class files because of its local classes.

Reply:
I'd like to add the functionality in the next issue. Thank you for the suggestion.